cpp/src/io/orc/writer_impl.cu

```diff
@@ -1237,19 +1240,20 @@ void writer::impl::write(table_view const &table)
     ff.types[0].subtypes.resize(num_columns);
     ff.types[0].fieldNames.resize(num_columns);
     for (auto const &column : orc_columns) {
-      ff.types[1 + column.id()].kind      = column.orc_kind();
-      ff.types[0].subtypes[column.id()]   = 1 + column.id();
-      ff.types[0].fieldNames[column.id()] = column.orc_name();
+      ff.types[1 + column.index()].kind      = column.orc_kind();
+      ff.types[0].subtypes[column.index()]   = 1 + column.index();
```

Comment:
What's the intent here?  Storing `(1 + column.index())`, or storing the id as understood by the Stream class?  If it's the latter, it might make sense to have a `.id()` accessor in the `orc_column_view` as well.

Reply:
good catch.